manager/common/license.go

```diff
@@ -120,8 +122,17 @@ type License struct {
 	Used  int
 }
 
-var totalCadenceLic = 3
-var totalXilinxLic = 3
+func strToInt(num string) int {
+	result, err := strconv.Atoi(num)
+	if err != nil {
+		log.Fatalf("Failed to convert %s to int", num)
+	}
+	return result
+}
+
+// Get total number of licenses from container env var
+var totalCadenceLic = strToInt(os.Getenv("CADENCE_LICENSES"))
+var totalXilinxLic = strToInt(os.Getenv("XILINX_LICENSES"))
```

Comment:
so... to fix the immediate issue, we just need to increase `totalXilinxLic` and `totalCadenceLic`, and restart the server. I recommend you do that, so we can move forward.

In terms of merging this code in the repository: I would strongly prefer not to have environment variables as the default config mechanism. Either command line flags, or a simple json / yaml config file.

The json/yaml file could load pretty much the entire content of LicenseCounter in one go. It would be pretty simple to do:
```
type LicenseConfig struct {
  Licenses map[string]*License
}
```
and then:
```
var config LicenseConfig
err := marshal.UnmarshalFile("path/to/config/file", &config)
```
with marshal library being:
`import "github.com/enfabrica/enkit/lib/config/marshal"`